### PR TITLE
[Zephyr] Fix ThreadMetrics diagnostic cluster implementation

### PR DESCRIPTION
**Pull request by axelnxp**
Addressing review comments from #36749

cc @Damian-Nordic

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]
Thank you!

File: src/platform/Zephyr/DiagnosticDataProviderImpl.cpp [hunk 1]
```diff
@@ -61,11 +61,16 @@ namespace {
 static void GetThreadInfo(const struct k_thread * thread, void * user_data)
 {
     size_t unusedStackSize;
-    ThreadMetrics ** threadMetricsListHead = (ThreadMetrics **) user_data;
-    ThreadMetrics * threadMetrics          = (ThreadMetrics *) malloc(sizeof(ThreadMetrics));
+    ThreadMetrics ** threadMetricsListHead = static_cast<ThreadMetrics **>(user_data);
+    ThreadMetrics * threadMetrics          = Platform::New<ThreadMetrics>();
```
Comment (Damian-Nordic) [inline src/platform/Zephyr/DiagnosticDataProviderImpl.cpp:65]: Actually, `DiagnosticDataProviderImpl::ReleaseThreadMetrics` should also use `Platform::Delete`

**Review by Damian-Nordic — APPROVED** [review]

File: src/platform/Zephyr/DiagnosticDataProviderImpl.cpp (hunk 1 above)
Comment (axelnxp) [reply src/platform/Zephyr/DiagnosticDataProviderImpl.cpp:65]: I was wondering how the memory was released  actually, you gave me the answer :) I'll fix that

Comment (axelnxp) [reply src/platform/Zephyr/DiagnosticDataProviderImpl.cpp:65]: fixed in the last push

**Review by chapongatien — APPROVED** [review]

**Pull request merged**